src/agent/configuration_parser/include/configuration_parser.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <toml.hpp>
+
+#include <iostream>
+
+namespace configuration
+{
+    class ConfigurationParser
+    {
+    private:
+        std::string configFileName = "wazuh.conf";
+        toml::value tbl;
+
+    public:
+        ConfigurationParser();
+        ConfigurationParser(std::string stringToParse);
+        ~ConfigurationParser() {};
+
+        template<typename T, typename... Ks>
+        auto GetConfig(Ks... ks)
+        {
+            try
+            {
+                auto config = toml::find<T>(tbl, ks...);
+                return config;
+            }
+            catch (const std::exception& e)
+            {
+                std::cout << "The requested value could not be obtained: " << e.what() << std::endl;
```

Comment:
This should be replaced by a logger once defined.

Reply:
Should we create a story to create such logger?